skimage/measure/fit.py

```diff
@@ -785,54 +790,64 @@ def ransac(data, model_class, min_samples, residual_threshold,
 
     random_state = check_random_state(random_state)
 
-    if min_samples < 0:
-        raise ValueError("`min_samples` must be greater than zero")
+    # in case data is not pair of input and output, male it like it
+    if not isinstance(data, (tuple, list)):
+        data = (data, )
+    data_lens = list(map(len, data))
+    assert len(set(data_lens)) == 1, 'all data have to have the same number of samples,' \
+                                     ' provided: %r' % [d.shape for d in data]
+    num_samples = data_lens[0]
+
+    if not (0 < min_samples <= num_samples):
+        raise ValueError("`min_samples` must be in range (0, <number-of-samples>]")
```

Comment:
This contrasts to the documentation, which states that it must be in `[0, <nb-samples>)`.

Reply:
it is in the documentation or what do you mean
```
min_samples : int in range [0, <nb-samples>)
        The minimum number of data points to fit a model to.
```

Reply:
btw, does it make sense to have there all samples? then RANSAC does not have any sense, right? so just (0, <nb-samples>)... @jni